Issues with exporting models on Windows

<!--
**Reminders**
1. This is not a place to debug your MCNP models
1. MCNP is export controlled and only its 6.2 and 6.3 user manuals are public. Don't include any information about MCNP which is not in those manuals.
1. Your model may be export controlled, or proprietary. Please change specific numbers (e.g., dimensions, material compositions, etc.) in your minimum working examples. -->

## Bug Description

This bug came up while testing #743: The test suite when ran on windows fails. Specifically it appears to be the tests that test exporting an MCNP model. Some of the failures are benign, like shortcuts not recompressing, but some of the test failures suggest a lost of information. 


## To Reproduce

<!-- A short code snippet of what you have ran. Please change or remove any specific values or anything that can't be public. For example: --> 

On windows run: 
``` bash
pytest
```

## Error Message 

<!-- If an error message was printed please include the entire stacktrace. If it includes any specific values please change or remove them. For example: -->

<details > 

``` python
============================= test session starts =============================
platform win32 -- Python 3.13.2, pytest-8.3.5, pluggy-1.5.0
rootdir: D:\a\MontePy\MontePy
configfile: pyproject.toml
plugins: hypothesis-6.131.0, profiling-1.8.1
collected 95 items

tests\test_integration.py ............F................................. [ 48%]
......................................F......FF..                        [100%]

================================== FAILURES ===================================
______________________________ test_problem_str _______________________________

simple_problem = MCNP problem for: tests\inputs\test.imcnp
MESSAGE: 3 lines
TITLE: MCNP Test Model for MOAA
CELL: 1, mat: 1, DENS: 20.0...src 
DATA INPUT: kcode 
DATA INPUT: phys:p 
Mode: {<Particle.NEUTRON: 'N'>, <Particle.PHOTON: 'P'>}
vol NO 2J 1 1.5 J



    def test_problem_str(simple_problem):
        output = str(simple_problem)
>       assert "MCNP problem for: tests/inputs/test.imcnp" in output
E       AssertionError: assert 'MCNP problem for: tests/inputs/test.imcnp' in 'MCNP problem for: tests\\inputs\\test.imcnp, TITLE: MCNP Test Model for MOAA'

tests\test_integration.py:190: AssertionError
________________________ test_read_write_cycle[file6] _________________________

file = WindowsPath('tests/inputs/test_interp_edge.imcnp')

    @pytest.mark.parametrize(
        "file",
        set((Path("tests") / "inputs").iterdir())
        - {
            Path("tests")
            / "inputs"
            / p  #                           Skip complexity of read
            for p in constants.BAD_INPUTS
            | constants.IGNORE_FILES
            | {"testRead.imcnp", "readEdgeCase.imcnp"}
        },
    )
    def test_read_write_cycle(file):
        print(f"Testing against {file} *********************")
        if ".swp" in file.suffixes:
            return
        problem = montepy.read_input(file, multi_proc=MULTI_PROC)
        SKIPPERS = _SKIP_LINES.get(str(file), {})
        fh = io.StringIO()
        # make string unclosable to keep open after reading.
        fh.close = lambda: None
        problem.write_problem(fh)
        fh.seek(0)
        # test valid syntax
        new_problem = montepy.read_input(fh, multi_proc=MULTI_PROC)
        # verify lines are similar
        fh.seek(0)
        lines = [line.rstrip() for line in fh]
        [print(line) for line in lines]
        with open(file, "r") as gold_fh:
            gold_fh_iter = iter(gold_fh)
            lines_iter = iter(lines)
            for i, (gold_line, new_line) in enumerate(zip(gold_fh_iter, lines_iter)):
                if i in SKIPPERS:
                    # True means skip new file line
                    if SKIPPERS[i] == 1:
                        new_line = next(lines_iter)
                    elif SKIPPERS[i] == 0:
                        continue
                    else:
                        gold_line = next(gold_fh_iter)
                # edge case override for not fixing #527.
                if str(file) == "tests/inputs/test_interp_edge.imcnp" and i == 1:
                    assert new_line == "10214   0    (1  2I 4 )"
                    continue
                try:
                    assert new_line == gold_line.rstrip().expandtabs(8)
                except AssertionError as e:
                    # handle case of making importance explicit
                    if "IMP:n=0.0" in new_line:
                        assert (
                            new_line.replace("IMP:n=0.0", "").rstrip() == gold_line.rstrip()
                        )
                    else:
>                       raise e

tests\test_integration.py:[12](https://github.com/idaholab/MontePy/actions/runs/14449013152/job/40516732086?pr=743#step:6:13)49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

file = WindowsPath('tests/inputs/test_interp_edge.imcnp')

    @pytest.mark.parametrize(
        "file",
        set((Path("tests") / "inputs").iterdir())
        - {
            Path("tests")
            / "inputs"
            / p  #                           Skip complexity of read
            for p in constants.BAD_INPUTS
            | constants.IGNORE_FILES
            | {"testRead.imcnp", "readEdgeCase.imcnp"}
        },
    )
    def test_read_write_cycle(file):
        print(f"Testing against {file} *********************")
        if ".swp" in file.suffixes:
            return
        problem = montepy.read_input(file, multi_proc=MULTI_PROC)
        SKIPPERS = _SKIP_LINES.get(str(file), {})
        fh = io.StringIO()
        # make string unclosable to keep open after reading.
        fh.close = lambda: None
        problem.write_problem(fh)
        fh.seek(0)
        # test valid syntax
        new_problem = montepy.read_input(fh, multi_proc=MULTI_PROC)
        # verify lines are similar
        fh.seek(0)
        lines = [line.rstrip() for line in fh]
        [print(line) for line in lines]
        with open(file, "r") as gold_fh:
            gold_fh_iter = iter(gold_fh)
            lines_iter = iter(lines)
            for i, (gold_line, new_line) in enumerate(zip(gold_fh_iter, lines_iter)):
                if i in SKIPPERS:
                    # True means skip new file line
                    if SKIPPERS[i] == 1:
                        new_line = next(lines_iter)
                    elif SKIPPERS[i] == 0:
                        continue
                    else:
                        gold_line = next(gold_fh_iter)
                # edge case override for not fixing #527.
                if str(file) == "tests/inputs/test_interp_edge.imcnp" and i == 1:
                    assert new_line == "10214   0    (1  2I 4 )"
                    continue
                try:
>                   assert new_line == gold_line.rstrip().expandtabs(8)
E                   AssertionError: assert '10214   0    (1  2I 4 )' == '10214   0    (1  2i  4)'
E                     
E                     - 10214   0    (1  2i  4)
E                     ?                   ^^
E                     + 10214   0    (1  2I 4 )
E                     ?                   ^  +

tests\test_integration.py:1241: AssertionError
---------------------------- Captured stdout call -----------------------------
Testing against tests\inputs\test_interp_edge.imcnp *********************
Surface interpolate edge case
10214   0    (1  2I 4 )

1 CZ 0
2 CZ 0
3 CZ 0
4 CZ 0

imp:n 0


________________________ test_read_write_cycle[file[13](https://github.com/idaholab/MontePy/actions/runs/14449013152/job/40516732086?pr=743#step:6:14)] ________________________

file = WindowsPath('tests/inputs/test_universe_data.imcnp')

    @pytest.mark.parametrize(
        "file",
        set((Path("tests") / "inputs").iterdir())
        - {
            Path("tests")
            / "inputs"
            / p  #                           Skip complexity of read
            for p in constants.BAD_INPUTS
            | constants.IGNORE_FILES
            | {"testRead.imcnp", "readEdgeCase.imcnp"}
        },
    )
    def test_read_write_cycle(file):
        print(f"Testing against {file} *********************")
        if ".swp" in file.suffixes:
            return
        problem = montepy.read_input(file, multi_proc=MULTI_PROC)
        SKIPPERS = _SKIP_LINES.get(str(file), {})
        fh = io.StringIO()
        # make string unclosable to keep open after reading.
        fh.close = lambda: None
        problem.write_problem(fh)
        fh.seek(0)
        # test valid syntax
        new_problem = montepy.read_input(fh, multi_proc=MULTI_PROC)
        # verify lines are similar
        fh.seek(0)
        lines = [line.rstrip() for line in fh]
        [print(line) for line in lines]
        with open(file, "r") as gold_fh:
            gold_fh_iter = iter(gold_fh)
            lines_iter = iter(lines)
            for i, (gold_line, new_line) in enumerate(zip(gold_fh_iter, lines_iter)):
                if i in SKIPPERS:
                    # True means skip new file line
                    if SKIPPERS[i] == 1:
                        new_line = next(lines_iter)
                    elif SKIPPERS[i] == 0:
                        continue
                    else:
                        gold_line = next(gold_fh_iter)
                # edge case override for not fixing #527.
                if str(file) == "tests/inputs/test_interp_edge.imcnp" and i == 1:
                    assert new_line == "102[14](https://github.com/idaholab/MontePy/actions/runs/14449013152/job/40516732086?pr=743#step:6:15)   0    (1  2I 4 )"
                    continue
                try:
>                   assert new_line == gold_line.rstrip().expandtabs(8)
E                   AssertionError: assert '     IMP:n=0.0' == '2 2 8'
E                     
E                     - 2 2 8
E                     +      IMP:n=0.0

tests\test_integration.py:1241: AssertionError

During handling of the above exception, another exception occurred:

file = WindowsPath('tests/inputs/test_universe_data.imcnp')

    @pytest.mark.parametrize(
        "file",
        set((Path("tests") / "inputs").iterdir())
        - {
            Path("tests")
            / "inputs"
            / p  #                           Skip complexity of read
            for p in constants.BAD_INPUTS
            | constants.IGNORE_FILES
            | {"testRead.imcnp", "readEdgeCase.imcnp"}
        },
    )
    def test_read_write_cycle(file):
        print(f"Testing against {file} *********************")
        if ".swp" in file.suffixes:
            return
        problem = montepy.read_input(file, multi_proc=MULTI_PROC)
        SKIPPERS = _SKIP_LINES.get(str(file), {})
        fh = io.StringIO()
        # make string unclosable to keep open after reading.
        fh.close = lambda: None
        problem.write_problem(fh)
        fh.seek(0)
        # test valid syntax
        new_problem = montepy.read_input(fh, multi_proc=MULTI_PROC)
        # verify lines are similar
        fh.seek(0)
        lines = [line.rstrip() for line in fh]
        [print(line) for line in lines]
        with open(file, "r") as gold_fh:
            gold_fh_iter = iter(gold_fh)
            lines_iter = iter(lines)
            for i, (gold_line, new_line) in enumerate(zip(gold_fh_iter, lines_iter)):
                if i in SKIPPERS:
                    # True means skip new file line
                    if SKIPPERS[i] == 1:
                        new_line = next(lines_iter)
                    elif SKIPPERS[i] == 0:
                        continue
                    else:
                        gold_line = next(gold_fh_iter)
                # edge case override for not fixing #527.
                if str(file) == "tests/inputs/test_interp_edge.imcnp" and i == 1:
                    assert new_line == "10214   0    (1  2I 4 )"
                    continue
                try:
                    assert new_line == gold_line.rstrip().expandtabs(8)
                except AssertionError as e:
                    # handle case of making importance explicit
                    if "IMP:n=0.0" in new_line:
>                       assert (
                            new_line.replace("IMP:n=0.0", "").rstrip() == gold_line.rstrip()
                        )
E                       AssertionError: assert '' == '2 2 8'
E                         
E                         - 2 2 8

tests\test_integration.py:1245: AssertionError
---------------------------- Captured stdout call -----------------------------
Testing against tests\inputs\test_universe_data.imcnp *********************
MCNP Test Model for MOAA
C cells
c
1 1 20
         -1000  $ dollar comment
     IMP:n=0.0
2 2 8
      -1005
     imp:n=1
     imp:p=0.5 $ graveyard
3 3 -1
      1000 1005 -1010
c
c foo end comment
c
     IMP:n=0.0
99 0 1000:-1005 1010 $graveyard
     IMP:n=0.0
5 0
      #99 IMP:n=0.0

C surfaces
1000 SO 1
1005 RCC 0 1.5 -0.5 0 0 1 0.25
1010 SO 3

C data
C materials
C UO2 5 atpt enriched
u 350 2J -1
FILL 2J 350
m1        92235.80c           5 &
92238.80c          95
C Iron
m2        26054.80c        5.85
          26056.80c       91.75
          26057.80c        2.12
          26058.80c        0.28
C water
m3        1001.80c           2
           8016.80c           1
MT3 lwtr.23t
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
phys:p j 1 2j 1
mode n p
vol NO 2J 1 1.5 J


________________________ test_read_write_cycle[file14] ________________________

file = WindowsPath('tests/inputs/test_complement_edge.imcnp')

    @pytest.mark.parametrize(
        "file",
        set((Path("tests") / "inputs").iterdir())
        - {
            Path("tests")
            / "inputs"
            / p  #                           Skip complexity of read
            for p in constants.BAD_INPUTS
            | constants.IGNORE_FILES
            | {"testRead.imcnp", "readEdgeCase.imcnp"}
        },
    )
    def test_read_write_cycle(file):
        print(f"Testing against {file} *********************")
        if ".swp" in file.suffixes:
            return
        problem = montepy.read_input(file, multi_proc=MULTI_PROC)
        SKIPPERS = _SKIP_LINES.get(str(file), {})
        fh = io.StringIO()
        # make string unclosable to keep open after reading.
        fh.close = lambda: None
        problem.write_problem(fh)
        fh.seek(0)
        # test valid syntax
        new_problem = montepy.read_input(fh, multi_proc=MULTI_PROC)
        # verify lines are similar
        fh.seek(0)
        lines = [line.rstrip() for line in fh]
        [print(line) for line in lines]
        with open(file, "r") as gold_fh:
            gold_fh_iter = iter(gold_fh)
            lines_iter = iter(lines)
            for i, (gold_line, new_line) in enumerate(zip(gold_fh_iter, lines_iter)):
                if i in SKIPPERS:
                    # True means skip new file line
                    if SKIPPERS[i] == 1:
                        new_line = next(lines_iter)
                    elif SKIPPERS[i] == 0:
                        continue
                    else:
                        gold_line = next(gold_fh_iter)
                # edge case override for not fixing #527.
                if str(file) == "tests/inputs/test_interp_edge.imcnp" and i == 1:
                    assert new_line == "10214   0    (1  2I 4 )"
                    continue
                try:
                    assert new_line == gold_line.rstrip().expandtabs(8)
                except AssertionError as e:
                    # handle case of making importance explicit
                    if "IMP:n=0.0" in new_line:
                        assert (
                            new_line.replace("IMP:n=0.0", "").rstrip() == gold_line.rstrip()
                        )
                    else:
>                       raise e

tests\test_integration.py:1249: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

file = WindowsPath('tests/inputs/test_complement_edge.imcnp')

    @pytest.mark.parametrize(
        "file",
        set((Path("tests") / "inputs").iterdir())
        - {
            Path("tests")
            / "inputs"
            / p  #                           Skip complexity of read
            for p in constants.BAD_INPUTS
            | constants.IGNORE_FILES
            | {"testRead.imcnp", "readEdgeCase.imcnp"}
        },
    )
    def test_read_write_cycle(file):
        print(f"Testing against {file} *********************")
        if ".swp" in file.suffixes:
            return
        problem = montepy.read_input(file, multi_proc=MULTI_PROC)
        SKIPPERS = _SKIP_LINES.get(str(file), {})
        fh = io.StringIO()
        # make string unclosable to keep open after reading.
        fh.close = lambda: None
        problem.write_problem(fh)
        fh.seek(0)
        # test valid syntax
        new_problem = montepy.read_input(fh, multi_proc=MULTI_PROC)
        # verify lines are similar
        fh.seek(0)
        lines = [line.rstrip() for line in fh]
        [print(line) for line in lines]
        with open(file, "r") as gold_fh:
            gold_fh_iter = iter(gold_fh)
            lines_iter = iter(lines)
            for i, (gold_line, new_line) in enumerate(zip(gold_fh_iter, lines_iter)):
                if i in SKIPPERS:
                    # True means skip new file line
                    if SKIPPERS[i] == 1:
                        new_line = next(lines_iter)
                    elif SKIPPERS[i] == 0:
                        continue
                    else:
                        gold_line = next(gold_fh_iter)
                # edge case override for not fixing #527.
                if str(file) == "tests/inputs/test_interp_edge.imcnp" and i == 1:
                    assert new_line == "10214   0    (1  2I 4 )"
                    continue
                try:
>                   assert new_line == gold_line.rstrip().expandtabs(8)
E                   AssertionError: assert 'm813    1001...s in material' == 'm813    1001...s in material'
E                     
E                     - m813    1001 3r 2004 0.1 $ valid syntax to test shortcuts in material
E                     ?              ^^
E                     + m813    1001 1001 1001 1001 2004 0.1 $ valid syntax to test shortcuts in material
E                     ?              ^^^^^^^^^^^^^^

tests\test_integration.py:1241: AssertionError
---------------------------- Captured stdout call -----------------------------
Testing against tests\inputs\test_complement_edge.imcnp *********************
Edge case of complement inside of an or
61440 0
      -1 IMP:N=1
C Steel dummy spacer above and below outer capsule
61461 0
      2 -3 (4 : -5)
     U=6000 IMP:N=1
C Bounding water outer capsule
61482 0
      ( 6 #61440 ) : (#61461 ( 7: -8))
     U=6000 IMP:N=1
      imp:+,-= 0
61441 814 0.087
      (-61867 61870 -61759):(-61868 61869 -61759): (-61870 61868 -61759 61758)
      IMP:N=0
      imp:+ = 1

1 SO 5.0
2 PZ 0.0
3 PZ 0.1
4 PZ 0.2
5 PZ 0.4
6 PZ 0.5
7 PZ 0.6
8 PZ 0.8
61867 PZ 0.0
61759 PZ 0.0
61868 PZ 0.0
61869 PZ 0.0
61870 PZ 0.0
61758 PZ 0.0

m814    26000.55c 5.657-2
        42000.51c 1.260-3
        28000.50c 9.881-3
        24000.50c 1.581-2
        25055.51c 1.760-3
m14     1001 1.0
m813    1001 1001 1001 1001 2004 0.1 $ valid syntax to test shortcuts in material
m8[15](https://github.com/idaholab/MontePy/actions/runs/14449013152/job/40516732086?pr=743#step:6:16)    8016 1.0 1001 1001 1001 1001 2004 0.1 $ valid syntax to test shortcuts in material
sdef cel=2 pos=0 0 0 rad=d3 ext=d3 axs=0 0 1
mode n + -


=========================== short test summary info ===========================
FAILED tests/test_integration.py::test_problem_str - AssertionError: assert 'MCNP problem for: tests/inputs/test.imcnp' in 'MCNP problem for: tests\\inputs\\test.imcnp, TITLE: MCNP Test Model for MOAA'
FAILED tests/test_integration.py::test_read_write_cycle[file6] - AssertionError: assert '10214   0    (1  2I 4 )' == '10214   0    (1  2i  4)'
  
  - 10214   0    (1  2i  4)
  ?                   ^^
  + 10214   0    (1  2I 4 )
  ?                   ^  +
FAILED tests/test_integration.py::test_read_write_cycle[file13] - AssertionError: assert '' == '2 2 8'
  
  - 2 2 8
FAILED tests/test_integration.py::test_read_write_cycle[file14] - AssertionError: assert 'm813    1001...s in material' == 'm813    1001...s in material'
  
  - m813    1001 3r 2004 0.1 $ valid syntax to test shortcuts in material
  ?              ^^
  + m813    1001 1001 1001 1001 2004 0.1 $ valid syntax to test shortcuts in material
  ?              ^^^^^^^^^^^^^^
======================== 4 failed, 91 passed in 2.[16](https://github.com/idaholab/MontePy/actions/runs/14449013152/job/40516732086?pr=743#step:6:17)s =========================
```

</details>

## Version

 - Version 1.0.0a5.dev48+gae2fc7f8

## Additional context

This has really shattered my confidence in Python's portability. More cross-os testing is needed. 

## Workaround

* Install and use the windows Subsystem for Linux.

--- a/montepy/errors.py
+++ b/montepy/errors.py
@@ -71,7 +71,8 @@ class ParsingError(MalformedInputError):
         else:
             self.message = message
 
-        # ValueError.__init__(self, self.message)
+    def __str__(self):
+        return self.message
 
 
 def _print_input(

--- a/montepy/errors.py
+++ b/montepy/errors.py
@@ -173,7 +173,7 @@ class ParticleTypeNotInCell(ParticleTypeWarning):
     pass
 
 
-class UnsupportedFeature(Exception):
+class UnsupportedFeature(ParsingError):
     """Raised when MCNP syntax that is not supported is found"""
 
     def __init__(self, message, input=None, error_queue=None):

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -187,7 +187,7 @@ def test_read_card_recursion():
 
 def test_problem_str(simple_problem):
     output = str(simple_problem)
-    assert "MCNP problem for: tests/inputs/test.imcnp" in output
+    assert f"MCNP problem for: {simple_problem.input_file.name}" in output
 
 
 def test_write_to_file(simple_problem):

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -1186,9 +1186,9 @@ def test_alternate_encoding():
 
 _SKIP_LINES = {
     # skip lines of added implied importances
-    "tests/inputs/test_universe_data.imcnp": {5: 1, 14: 1, 15: 1},
+    Path("tests")/"inputs"/"test_universe_data.imcnp": {5: 1, 14: 1, 15: 1},
     # I don't care about the edge case of shortcuts in a material def.
-    "tests/inputs/test_complement_edge.imcnp": {37: 0, 38: 0, 39: 0},
+    Path("tests")/"inputs"/"test_complement_edge.imcnp": {37: 0, 38: 0, 39: 0},
 }
 
 
@@ -1209,7 +1209,7 @@ def test_read_write_cycle(file):
     if ".swp" in file.suffixes:
         return
     problem = montepy.read_input(file, multi_proc=MULTI_PROC)
-    SKIPPERS = _SKIP_LINES.get(str(file), {})
+    SKIPPERS = _SKIP_LINES.get(file, {})
     fh = io.StringIO()
     # make string unclosable to keep open after reading.
     fh.close = lambda: None
@@ -1234,7 +1234,7 @@ def test_read_write_cycle(file):
                 else:
                     gold_line = next(gold_fh_iter)
             # edge case override for not fixing #527.
-            if str(file) == "tests/inputs/test_interp_edge.imcnp" and i == 1:
+            if file == Path("tests")/"inputs"/"test_interp_edge.imcnp" and i == 1:
                 assert new_line == "10214   0    (1  2I 4 )"
                 continue
             try:

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -1186,9 +1186,9 @@ def test_alternate_encoding():
 
 _SKIP_LINES = {
     # skip lines of added implied importances
-    Path("tests")/"inputs"/"test_universe_data.imcnp": {5: 1, 14: 1, 15: 1},
+    Path("tests") / "inputs" / "test_universe_data.imcnp": {5: 1, 14: 1, 15: 1},
     # I don't care about the edge case of shortcuts in a material def.
-    Path("tests")/"inputs"/"test_complement_edge.imcnp": {37: 0, 38: 0, 39: 0},
+    Path("tests") / "inputs" / "test_complement_edge.imcnp": {37: 0, 38: 0, 39: 0},
 }
 
 
@@ -1234,7 +1234,7 @@ def test_read_write_cycle(file):
                 else:
                     gold_line = next(gold_fh_iter)
             # edge case override for not fixing #527.
-            if file == Path("tests")/"inputs"/"test_interp_edge.imcnp" and i == 1:
+            if file == Path("tests") / "inputs" / "test_interp_edge.imcnp" and i == 1:
                 assert new_line == "10214   0    (1  2I 4 )"
                 continue
             try:

